tests/src/test/scala/system/basic/WskRestBasicTests.scala

```diff
@@ -444,16 +480,36 @@ class WskRestBasicTests extends TestHelpers with WskTestHelpers with WskActorSys
 
     result.getField("name") shouldBe name
     result.getField("namespace") shouldBe ns
+
     val annos = result.getFieldJsValue("annotations")
-    annos shouldBe JsArray(
-      JsObject("key" -> JsString("description"), "value" -> JsString("Action description")),
-      JsObject(
-        "key" -> JsString("parameters"),
-        "value" -> JsArray(
-          JsObject("name" -> JsString("paramName1"), "description" -> JsString("Parameter description 1")),
-          JsObject("name" -> JsString("paramName2"), "description" -> JsString("Parameter description 2")))),
-      JsObject("key" -> WhiskAction.provideApiKeyAnnotationName.toJson, "value" -> JsBoolean(false)),
-      JsObject("key" -> JsString("exec"), "value" -> JsString("nodejs:6")))
+    annos shouldBe (if (requireAPIKeyAnnotation) {
+                      JsArray(
+                        JsObject("key" -> JsString("description"), "value" -> JsString("Action description")),
+                        JsObject(
+                          "key" -> JsString("parameters"),
+                          "value" -> JsArray(
+                            JsObject(
+                              "name" -> JsString("paramName1"),
+                              "description" -> JsString("Parameter description 1")),
+                            JsObject(
+                              "name" -> JsString("paramName2"),
+                              "description" -> JsString("Parameter description 2")))),
+                        JsObject("key" -> WhiskAction.provideApiKeyAnnotationName.toJson, "value" -> JsBoolean(false)),
+                        JsObject("key" -> JsString("exec"), "value" -> JsString("nodejs:6")))
+                    } else {
+                      JsArray(
+                        JsObject("key" -> JsString("description"), "value" -> JsString("Action description")),
+                        JsObject(
+                          "key" -> JsString("parameters"),
+                          "value" -> JsArray(
+                            JsObject(
+                              "name" -> JsString("paramName1"),
+                              "description" -> JsString("Parameter description 1")),
+                            JsObject(
+                              "name" -> JsString("paramName2"),
+                              "description" -> JsString("Parameter description 2")))),
+                        JsObject("key" -> JsString("exec"), "value" -> JsString("nodejs:6")))
```

Comment:
I think the annotations comment applies to all the instances you touched annotation lists.